packages/wallet/test/integration/buildTx.test.ts

```diff
@@ -0,0 +1,143 @@
+    assertTxIsValid(tx);
+    // 2 outputs + change output
+    expect(tx.body.outputs.length).toBe(3);
+  });
+
+  it('can remove outputs and build', async () => {
+    const txBuilder = buildTx(wallet);
+    const txTwoOutputs = txBuilder.addOutput(mocks.utxo[0][1]).addOutput(mocks.utxo[1][1]);
+    const tx = await txTwoOutputs.removeOutput(mocks.utxo[0][1]).build();
+    assertTxIsValid(tx);
+    expect(tx.body.outputs.length).toBe(2);
+  });
+
+  it('builds transactions that are not modified by subsequent builder changes', async () => {
+    const txBuilder = buildTx(wallet);
+    const tx = await txBuilder.addOutput(mocks.utxo[0][1]).build();
+    await txBuilder.addOutput(mocks.utxo[1][1]).build();
+
+    assertTxIsValid(tx);
+    expect(tx.body.outputs.length).toBe(2);
+  });
+
+  it('validates multiple outputs before building', async () => {
+    const coinMissingValidation: OutputValidation = {
+      coinMissing: 1n,
+      minimumCoin: 2n,
+      tokenBundleSizeExceedsLimit: false
+    };
+
+    const bundleSizeValidation: OutputValidation = {
+      coinMissing: 0n,
+      minimumCoin: 2n,
+      tokenBundleSizeExceedsLimit: true
+    };
+
+    const output1 = mocks.utxo[0][1];
+    const output2 = mocks.utxo[1][1];
+
+    const mockValidator: OutputValidator = {
+      validateOutput: jest.fn(),
+      validateOutputs: jest.fn(() =>
+        Promise.resolve(
+          new Map([
+            [output1, coinMissingValidation],
+            [output2, bundleSizeValidation]
+          ])
+        )
+      ),
+      validateValue: jest.fn(),
+      validateValues: jest.fn()
+    };
+    const tx = await buildTx(wallet, mockValidator).addOutput(output1).addOutput(output2).build();
+
+    expect(tx.isValid).toBeFalsy();
+    const [error1, error2] = (!tx.isValid && tx.errors) || [];
+
+    expect(error1 instanceof OutputValidationError).toBeTruthy();
+    if (error1 instanceof OutputValidationError) {
+      expect(error1.failure === TxOutputFailure.coinMissing);
+    }
+
+    expect(error2 instanceof OutputValidationError).toBeTruthy();
+    if (error2 instanceof OutputValidationError) {
+      expect(error2.failure === TxOutputFailure.tokenBundleSizeExceedsLimit);
+    }
+  });
+
+  describe('buildOutput', () => {
+    it.skip('can build a valid output', async () => {
+      const assetId = Cardano.AssetId('1ec85dcee27f2d90ec1f9a1e4ce74a667dc9be8b184463223f9c960150584c');
+      const assetQuantity = 100n;
+      const assets = new Map([[assetId, assetQuantity]]);
+      const address = Cardano.Address('addr_test1vr8nl4u0u6fmtfnawx2rxfz95dy7m46t6dhzdftp2uha87syeufdg');
+      const datum = Cardano.Hash32ByteBase16('3e33018e8293d319ef5b3ac72366dd28006bd315b715f7e7cfcbd3004129b80d');
+      const output1Coin = 10_000_000n;
+      const output2Base = mocks.utxo[0][1];
+
+      const txBuilder = buildTx(wallet);
+
+      const output1 = await txBuilder
```

Comment:
Maybe it would be better communicated as:

``` ts
const output1 = await txBuilder.buildOutput();
      output1
        .address(address)
        .coin(output1Coin)
        .asset(assetId, assetQuantity)
        .build();
```

... since I read it the first time as "adding an address to a transaction"

Reply:
`build()` has to be awaited (while `buildOutput()` does not)

Reply:
[1d5d8f7](https://github.com/input-output-hk/cardano-js-sdk/pull/396/commits/1d5d8f785403e6238a2206d8be8673b8f3220f7e)